### PR TITLE
[CMOS-4055]: Remove deprecated prop checking from Plasma

### PR DESCRIPTION
**Pull request by chaitany6143**

### CODE REVIEW

**Review by kangax — COMMENTED** [review]
Instead of adding `/* eslint react/prop-types: 0 */` everywhere, we should just disable this rule everwhere in the project

File: src/components/Button/Button.jsx [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
+    children: Node,
+    style: Object,
+    isSubmit: boolean,
+    size: string,
+    data: Data,
+|};
+
+class Button extends React.Component<Props> {
+  static defaultProps = {
+    type: type.PRIMARY,
```
Comment (kangax) [inline src/components/Button/Button.jsx]: this actually works?!

File: src/components/Card/Card.jsx [hunk 2]
```diff
@@ -14,7 +16,28 @@ const cardTypes = {
 const borderTypes = { DASHED: 'dashed' };
 export const borderColors = { RED: 'red' };
 
-class Card extends React.Component {
+type Props = {|
+    title: string,
```
Comment (kangax) [inline src/components/Card/Card.jsx]: I think our convention is 2 spaces... speaking of which, we need prettier here :/

File: src/components/Card/Card.jsx [hunk 3]
```diff
@@ -25,16 +48,22 @@ class Card extends React.Component {
...
+          = ReactDOM.findDOMNode(this.expanded); //$FlowFixMe
       const expandedHeight = expandedEl.offsetHeight;
       this.setState({ defaultHeight, expandedHeight });
     }
     this.setState({ isMounted: true });
     /* eslint-enable */
   }
 
+  expanded: ?Object;
+  outer: ?Object;
```
Comment (kangax) [inline src/components/Card/Card.jsx]: I think these could be marked as Element?

File: src/components/Header/Header.jsx [hunk 4]
```diff
@@ -1,13 +1,28 @@
...
-  getDataAttrs,
-  getDataProps }
-from '../../dataUtils';
+import { getDataAttrs } from '../../dataUtils';
+import type { Data } from '../../types';
 
-class Header extends React.Component {
+type Props = {|
+    text?: string,
+    children: string,
```
Comment (kangax) [inline src/components/Header/Header.jsx]: children should always be `Node`

Comment (kangax) [reply src/components/Header/Header.jsx]: also could be optional

File: src/components/Table/Table.jsx [hunk 5]
```diff
@@ -1,28 +1,61 @@
...
+
+type State = {|
+    columnSizes: Object,
+    tableWidth: number,
+    backgroundColor?: ?string,
+    stickyCoverHeight?: number,
+    translateY?: number,
+    showSticky?: boolean,
+    headerHeight: number,
+
```
Comment (kangax) [inline src/components/Table/Table.jsx]: nit: extra newline

File: src/components/TextArea/TextArea.jsx [hunk 6]
```diff
@@ -1,20 +1,44 @@
...
 
-class TextArea extends Component {
+type Props = {|
+    autosize: boolean,
+    disabled: boolean,
+    error: boolean,
+    maxLength: string,
+    onBlur: () => void,
+    onChange: () => void,
+    onFocus: PropTypes.func,
```
Comment (kangax) [inline src/components/TextArea/TextArea.jsx]: plz get rid of PropTypes completely

File: src/components/TextInput/TextInput.jsx [hunk 7]
```diff
@@ -1,10 +1,28 @@
...
+    placeholder: string,
+    value: string,
+    onChange: () => void,
+    onFocus: () => void,
+    onBlur: () => void,
+    suffix: Node,
+    prefix: Node,
+    isDisabled: boolean,
+    disabled: boolean,
+    size: boolean,
```
Comment (kangax) [inline src/components/TextInput/TextInput.jsx]: shouldn't size be a string or a number?

File: src/components/Toggle/Toggle.jsx [hunk 8]
```diff
@@ -14,12 +13,31 @@ const type = {
   SMALL: 'small',
 };
 
-class Toggle extends React.Component {
-  onClick(toggle, index) {
+type Props = {|
+    items: {
+        label: string,
+        title: string,
```
Comment (kangax) [inline src/components/Toggle/Toggle.jsx]: incorrect spacing here as well

File: src/components/layout/FixedTop/FixedTop.jsx [hunk 9]
```diff
@@ -1,12 +1,27 @@
...
 import style from './style.scss';
 
-class FixedTop extends React.Component {
+
+type State = {|
+    fixedHeight: number,
+|};
+
+type Props = {|
+    children: Array<Node>,
```
Comment (kangax) [inline src/components/layout/FixedTop/FixedTop.jsx]: just `Node`

File: src/components/layout/Group/Group.jsx [hunk 10]
```diff
@@ -1,14 +1,25 @@
...
-  getDataAttrs,
-  getDataProps,
-} from '../../../dataUtils';
+import { getDataAttrs } from '../../../dataUtils';
+import type { Data } from '../../../types';
 import style from './style.scss';
 
-class Group extends React.Component {
+type Props = {|
+    children: Array<Node>,
```
Comment (kangax) [inline src/components/layout/Group/Group.jsx]: just `Node`

File: src/components/Button/Button.jsx (hunk 1 above)
Comment (chaitany6143) [reply src/components/Button/Button.jsx]: It does. But Changing it to buttonType for clarity.

File: src/components/Header/Header.jsx (hunk 4 above)
Comment (chaitany6143) [reply src/components/Header/Header.jsx]: oops. Changing.

File: src/components/layout/Group/Group.jsx (hunk 10 above)
Comment (chaitany6143) [reply src/components/layout/Group/Group.jsx]: Have to use Array<Node> as this.props.children.length is being accessed. Flow complains on using just Node.

**Review by kangax — APPROVED** [review]
🔥

**Pull request merged**